src/virtual_elements.js

```diff
@@ -275,13 +275,14 @@ var elementVoid = function(tag, key, statics, var_args) {
  * @param {...(function(string|number|boolean):string|number|boolean)} var_args
  *     Functions to format the value which are called only when the value has
  *     changed.
+ * @return {!Text} The corresponding Text Node.
  */
 var text = function(value, var_args) {
   if (process.env.NODE_ENV !== 'production') {
     assertNotInAttributes();
   }
 
-  var node = alignWithDOM('#text', null);
+  var node = /** @type {!Text}*/(alignWithDOM('#text', null));
```

Comment:
So firstly I'm not a fan of having these inline `@type`s either. You do need them if you are using the closure compiler because without them it complains about the type being a `Node`.

As I see it there seem to be a few solutions to this:
* Change the `text()` return to `{!Node}` which would remove typing information for anyone calling text
* Refactor `alignWithDOM` out to deal with Text and Element's differently which would then allow it to return the correct type from `createTextNode` directly
* Leave them off and ignore the closure compiler (if thats the case I would question why bothering with a partial type annotations in the first place)